lib/swiftner/API/service.rb

```diff
@@ -27,15 +27,29 @@ def self.validate_required(attributes, *keys)
         raise ArgumentError, "Key(s) '#{formatted_keys}' are missing in attributes. #{attributes.inspect}"
       end
 
-      def self.validate_language(attributes)
-        return unless attributes.key?(:language)
+      def self.validate_language(attributes, key = :language)
+        return unless attributes.key?(key)
 
-        return if Swiftner::Configuration::SUPPORTED_LANGUAGES.include?(attributes[:language])
+        return if Swiftner::Configuration::SUPPORTED_LANGUAGES.include?(attributes[key])
 
-        raise ArgumentError, "Language '#{attributes[:language]}' is not supported.
+        raise ArgumentError, "Language '#{attributes[key]}' is not supported.
                               Supported languages are #{Swiftner::Configuration::SUPPORTED_LANGUAGES.join(", ")}"
       end
 
+      def self.validate_file(file_path)
```

Comment:
Maybe we could use a library to detect the MIME type of a file instead of basing it of file extensions?

Reply:
Or maybe swiftner only returns file extensions?